core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2587,11 +2587,11 @@ class ReplicaManager(val config: KafkaConfig,
     new ReplicaAlterLogDirsManager(config, this, quotaManager, brokerTopicStats, directoryEventHandler)
   }
 
-  private def createReplicaSelector(): Option[ReplicaSelector] = {
+  private def createReplicaSelector(metrics: Metrics): Option[Plugin[ReplicaSelector]] = {
     config.replicaSelectorClassName.map { className =>
       val tmpReplicaSelector: ReplicaSelector = Utils.newInstance(className, classOf[ReplicaSelector])
       tmpReplicaSelector.configure(config.originals())
-      tmpReplicaSelector
+      Plugin.wrapInstance(tmpReplicaSelector, metrics, className)
```

Comment:
I just noticed that this is not correct. Instead of `className`, the second argument should be the name of the configuration: `REPLICA_SELECTOR_CLASS_CONFIG`.

I fixed that in my PR adding KIP-877 support to the authorizer: https://github.com/apache/kafka/pull/19050